### PR TITLE
Add docs detailing our ASF Jenkins usage

### PR DESCRIPTION
**Pull request by gerlowskija**
Hopefully these docs will answer a lot of questions I had when troubleshooting a recent outage of our ASF Jenkins agents.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by epugh — APPROVED** [review]
LGTM.  Thanks for adding this...    I wonder if we have 2 boxes because back int he day 1 for solr and 1 for lucene, and then they merged...?

File: dev-docs/asf-jenkins.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
+= ASF Jenkins Setup
+:toc: left
+
+The Solr project uses a Jenkins instance provided by the Apache Software Foundation ("ASF") for running tests, validation, etc.
+
+This file aims to document our ASF Jenkins usage and administrator, to prevent it from becoming "tribal knowledge" understood by just a few.
+
```
Comment (dsmiley) [inline dev-docs/asf-jenkins.adoc:7]: Let's include a link to Jenkins :-)

File: dev-docs/asf-jenkins.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+We run a number of jobs on Jenkins, each validating an overlapping set of concerns:
+
+* `Solr-Artifacts-*` - daily jobs that run `./gradlew assemble` to ensure that build artifacts (except docker images) can be created successfully
+* `Solr-check-*` - "hourly" jobs that run all project tests and static analysis (i.e. `test`, `integrationTest`, and `check`)
+* `Solr-Docker-Nightly-*` - daily jobs that `./gradlew testDocker dockerPush` to validate docker image packaging.  Snapshot images are pushed to hub.docker.com
+* `Solr-reference-guide-*` - hourly jobs that build the Solr reference guide via `./gradlew checkSite` and push the resulting artifact to the staging/preview site `nightlies.apache.org`
+* `Solr-Smoketest-*` - daily jobs that produce a snapshot release (via the `assembleRelease` task) and run the release smoketester
+
+Unless otherwise specified, all builds currently run using JDK 11.
```
Comment (dsmiley) [inline dev-docs/asf-jenkins.adoc:18]: Let's not write too much information here that could get out of sync with reality

File: dev-docs/asf-jenkins.adoc
Comment (dsmiley) [inline dev-docs/asf-jenkins.adoc:1]: Maybe rename to include "ci-" prefix?

**Review by dsmiley — COMMENTED** [review]
Thanks for writing this.

I'm almost surprised this went here instead of Confluence as it's easier to edit/collaborate on a wiki instead of in code and do PR process.

**Comment by epugh** [comment]
I view Confluence as a cesspool of out of date information.    We just don't pay attention to it the way we do with our `dev-docs`.    We *should* commit to one or the other.

**Review by janhoy — APPROVED** [review]
Thanks, this was helpful

**Comment by gerlowskija** [comment]
> I'm almost surprised this went here instead of Confluence as it's easier to edit/collaborate on a wiki instead of in code and do PR process.

I think that was probably true once, but it's at least arguably different now that Confluence account creation is semi-locked down as a defense against spam.  Committers all tend to have Confluence access already, but for non-committers I think there's a good case that GH collaboration has less friction.  I've helped a handful of folks get Confluence access this year (for updating SIPs, organizing "Community Meetups", etc.), and it's been a minor pain 🙁 

But Eric's already highlighted my main reason for going the "dev-docs" route - for better or worse, Confluence doesn't seem like it gets much attention any more.  Personally I'm OK with this - our user-facing documentation lives "in-repo" just fine, benefits from PR review, version control, etc.  Why would developer-facing docs be any different?

(This is probably worth a discussion on dev@, if anyone feels like injecting life back into use of cwiki as a documentation store.)

**Comment by gerlowskija** [comment]
Thanks all for the review - merged!

**Pull request merged**